src/Network/NWBrowser.cs

```diff
@@ -0,0 +1,152 @@
+//
+// NWBrowser.cs: Bindings the Network nw_browser_t API.
+//
+// Authors:
+//   Manuel de la Pena (mandel@microsoft.com)
+//
+// Copyrigh 2019 Microsoft Inc
+//
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+
+using OS_nw_browser=System.IntPtr;
+using OS_nw_browse_descriptor=System.IntPtr;
+using OS_nw_parameters=System.IntPtr;
+using dispatch_queue_t =System.IntPtr;
+
+namespace Network {
+
+	[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
+	public enum NWBrowserState {
+		Invalid = 0,
+		Ready = 1,
+		Failed = 2,
+		Cancelled = 3,
+	}
+
+	[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
+	public class NWBrowser : NativeObject {
+
+		public NWBrowser (IntPtr handle, bool owns) : base (handle, owns) {}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_browser nw_browser_create (OS_nw_browse_descriptor descriptor, OS_nw_parameters parameters);
+
+		public NWBrowser (NWBrowserDescriptor descriptor, NWParameters parameters)
+		{
+			if (descriptor == null)
+				throw new ArgumentNullException (nameof (descriptor));
+			var parametersHandler = parameters?.Handle ?? IntPtr.Zero;
+
+			InitializeHandle (nw_browser_create (descriptor.Handle, parameters.Handle));
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_browser_set_queue (OS_nw_browser browser, dispatch_queue_t queue);
+
+		public DispatchQueue DispatchQueue {
+			set {
+				if (value == null)
+					throw new ArgumentNullException (nameof (value));
+				nw_browser_set_queue (GetCheckedHandle (), value.Handle);
+			}
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_browser_start (OS_nw_browser browser);
+
+		public void Start () => nw_browser_start (GetCheckedHandle ());
```

Comment:
I know that this is not reflected in the Apple API, but if I have a `Start` and a `Cancel`, I'd like to have `bool IsStarted { get; private set; }` The Apple documentation doesn't say if you can call `Start` multiple times.